### ISSUE TITLE
RadialTimePickerDialogFragment AM/PM

### ISSUE THREAD

**Issue by paxxux** [code-troopers/android-betterpickers#243]
Hi,

The RadialTimePickerDialogFragment does not change to AM/PM if I set my device to non 24-Hour format.

It would be easy to verify, if the device is in 24-Hour mode or not:
boolean b = DateFormat.is24HourFormat(context);

I would suggest to fall back on the device time format, rather than setting it (mIs24HourMode = true;) to a default value.

I cannot see any possibility to override mIs24HourMode in RadialTimePickerDialogFragment.

Or do I just miss something?

Greetings,
paxxux

**Comment by fchauveau** [comment]
Hello @paxxux 

Indeed we actually do not correctly preset 24h-dateFormat
For now you can (must) set this param, (using is24HourFormat as you suggest) by calling 

```
RadialTimePickerDialogFragment#initialize(OnTimeSetListener callback, int hourOfDay, int minute, boolean is24HourMode)
```

We will soon release a new version with a better management of 24h-format,
thx

**Comment by paxxux** [comment]
Great! Thank you!

**Comment by fchauveau** [comment]
Hi @paxxux 
v2.4.0 have been release, you can now force detection on dateFormat using

```
RadialTimePickerDialogFragment.setAutodetectDateFormat(Context);
```

**Comment by paxxux** [comment]
Hi,

Thanks for the hint! Works great!

1) I would suggest to auto-detect by default. In my opinion people want a picker according to user preferences, rather than 24-hour format. It would make one's code more simple and your users do not have to search your API for auto-detection. I believe force 12/24-hour is a rarer use case.

2)

```
  RadialTimePickerDialogFragment.setAutodetectDateFormat(Context);
```

Why don't you use DialogFragment.getContext() instead of taking a Context argument? I am not really sure, but is there a case of more than one Context in one app?

**Comment by fchauveau** [comment]
1/ I didn't set it by default because I was waiting for Context 
2/ And I didn't tried to get context by myself.

But you are right DialogFragment.getContext() solve all of this.
So I'm going to autodetect by default using DialogFragment.getContext \o/

Thank you

**Comment by paxxux** [comment]
Thank you too :-)

**Comment by fchauveau** [comment]
@paxxux 
Correction is available in version 2.5.0 just released

**Comment by paxxux** [comment]
Great :) just updated. Works fine!
